src/content/post/launching-trieve-hn-discovery.mdx

```diff
@@ -0,0 +1,101 @@
+---
+publishDate: 2024-08-29T08:45:00.000Z
+author: skeptrune (Nick K)
+title: >-
+  Launching Trieve HN Discovery - A Discovery Focused Search Engine for Hacker
+  News
+excerpt: >-
+  Trieve HN Discovery is an open source discovery-focused search engine for
+  Hacker News. With features like LLM-generated query suggestions, site filters,
+  semantic search, RAG AI chat, and order by descendant count, it offers a
+  comprehensive search experience for the Hacker News community.
+image: >-
+  https://cdn.trieve.ai/blog/trieve-hn-discovery/trieve-hn-discovery-preview-opengraph.webp
+category: Demos
+tags:
+  - hackernews
+  - search
+  - demo
+---
+
+We have been interested in building a HN discovery interface since before we were accepted into the W24 Y-Combinator batch in late 2023\. However, we were limited by our own product not being good enough to develop what we wanted. Our ingest speed and reliability in particular were extreme challenges. Now, [after 6 months of hacking in our spare time](https://github.com/devflowinc/trieve-hn-discovery/commit/2e24998d30b336416a947ee1594eb9c026958087), we are excited to share the complete version available at [hn.trieve.ai](https://hn.trieve.ai)!
+
+Backend source code available at [github.com/devflowinc/trieve](https://github.com/devflowinc/trieve) and the frontend code at [github.com/devflowinc/trieve-hn-discovery](https://github.com/devflowinc/trieve-hn-discovery). See the [system diagram here](https://docs.trieve.ai/getting-started/introduction#system-diagram) for a high level overview.
+
+# New Features!
+
+Quote and negated word, author, story id, points, comments, and date filters are all supported. We thought it was extremely important to preserve all existing functionality.
+
+## No JS Version
+
+We have both [hnnojs.trieve.ai](https://hnnojs.trieve.ai/) and [hn.trieve.ai](https://hn.trieve.ai/). Respectively, they are a Rust [actix-web server](https://github.com/devflowinc/trieve-hn-discovery/tree/main/actix-frontend) with [minijinja](https://github.com/mitsuhiko/minijinja) templates and a [SolidJS SPA](https://github.com/devflowinc/trieve-hn-discovery/tree/main/frontend).
+
+A dedicated blog on how we built the no-js version in Rust will come out soon!
+
+## LLM Generated Query Suggestions
+
+We used our [generate suggested queries route](https://docs.trieve.ai/api-reference/chunk/generate-suggested-queries) to show query suggestions based on random posts from HN.
+
+## Site Filters
+
+Search results can now be limited on a URL basis! For example you can query for "async site:blog.rust-lang.org" to see discussion of async Rust specifically shared by the official rust blog.
+
+Negated filters like "async rust site:-blog.rust-lang.org" to exclude results from the official Rust blog also work.
```

Comment:
Note: This is distinct from how site: operators work in Google (`-site:`) and in how operators work on, like, Twitter (`-from:danielsgriffin`) thought it is somewhat more natural. Natural language here is: "site constrained to NOT xyz"